test/scholar/linear/polynomial_regression_test.exs

```diff
@@ -0,0 +1,298 @@
+defmodule Scholar.Linear.PolynomialRegressionTest do
+  use Scholar.Case, async: true
+  # doctest Scholar.Linear.PolynomialRegression
```

Comment:
Should this be uncommented?

Reply:
Yes, currently there is a compilation error that I can't figure out with those doctests
```
  1) doctest Scholar.Linear.PolynomialRegression.fit/3 (2) (Scholar.Linear.PolynomialRegressionTest)
     test/scholar/linear/polynomial_regression_test.exs:3
     Doctest did not compile, got: (SyntaxError) lib/scholar/linear/polynomial_regression.ex:91:5: syntax error before: '['
         |
      91 |     [-0.021396497264504433, -0.004854594357311726, -0.0884987860918045, -0.062211357057094574, -0.04369127005338669]
         |     ^
     doctest:
       iex> x = Nx.tensor([[1.0, 2.0], [3.0, 2.0], [4.0, 7.0]])
       iex> y = Nx.tensor([4.0, 3.0, -1.0])
       iex> Scholar.Linear.PolynomialRegression.fit(x, y, degree: 2)
       %Scholar.Linear.PolynomialRegression{
         coefficients: #Nx.Tensor<
           f32[5]
           [-0.021396497264504433, -0.004854594357311726, -0.0884987860918045, -0.062211357057094574, -0.04369127005338669]
         >,
         intercept: #Nx.Tensor<
           f32
           4.418517112731934
         >,
         degree: 2
       }
     stacktrace:
       lib/scholar/linear/polynomial_regression.ex:88: Scholar.Linear.PolynomialRegression (module)
```
Maybe you're familiar with this error and know how what could be the cause

Reply:
Yeah, you can't inspect with tensors. You can see other examples, we typically do this:

```
       iex> Scholar.Linear.PolynomialRegression.fit(x, y, degree: 2)
       %Scholar.Linear.PolynomialRegression{
         coefficients: Nx.tensor([-0.021396497264504433, -0.004854594357311726, -0.0884987860918045, -0.062211357057094574, -0.04369127005338669]),
         intercept: Nx.tensor(4.418517112731934),
         degree: 2
       }
```

Reply:
I just saw that linear regression had the same problem, I'll do that change for both

Reply:
Yup, your approach is also great. :)